llvm/lib/Transforms/Vectorize/VPlanTransforms.h

```diff
@@ -103,6 +103,12 @@ struct VPlanTransforms {
   /// not valid.
   static bool adjustFixedOrderRecurrences(VPlan &Plan, VPBuilder &Builder);
 
+  /// Check if \p Plan contains any FMaxNumNoFMFs or FMinNumNoFMFs reductions.
+  /// If they do, try to update the vector loop to exit early if any input is
+  /// NaN and resume executing in the scalar loop to handle the NaNs there.
+  /// Return false if this attempt was unsuccessful.
+  static bool handleMaxMinNumReductionsWithoutFastMath(VPlan &Plan);
```

Comment:
```suggestion
  static bool handleMaxMinNumReductions(VPlan &Plan);
```
(let the potential one who would rely on FMF say so?)